src/modules/kvs/kvstxn.c

```diff
@@ -914,10 +914,8 @@ kvstxn_process_t kvstxn_process (kvstxn_t *kt, const char *rootdir_ref)
         }
 
         if (kt->errnum != 0) {
-            char *ref;
             /* empty missing_refs_list to prevent mistakes later */
-            while ((ref = zlist_pop (kt->missing_refs_list)))
-                free (ref);
+            zlist_purge (kt->missing_refs_list);
```

Comment:
Need to set `zlits_autofree()`?

Reply:
I set `zlist_freefn()` in `add_missing_ref()`, so I think that'll be called on purge.  But if see what you're saying.  If I set `zlist_autofree()`, i wouldn't have to call `strdup` or `zlist_freefn()` at all.  Lemme clean that up.

Reply:
Oops I forgot to look for `zlist_freefn()`.  My bad.